pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -116,30 +101,51 @@ public static SqlNodeAndOptions compileToSqlNodeAndOptions(String sql)
     // Remove the terminating semicolon from the query
     sql = removeTerminatingSemicolon(sql);
 
-    // Extract OPTION statements from sql as Calcite Parser doesn't parse it.
-    List<String> options = extractOptionsFromSql(sql);
-    if (!options.isEmpty()) {
-      sql = removeOptionsFromSql(sql);
-    }
-
     try (StringReader inStream = new StringReader(sql)) {
```

Comment:
Not related to this PR --

Looks like this function was added when INSERT grammar extension was done and is being called from broker query endpoint in `PinotClientRequest.java`.

It seems like for DQL, the caller will parse the SQL statement twice. Once at line 153 when it calls this method in the parser to retrieve `SqlNodeAndOptions` and then if the type is not DML, it sends the query to `requestHandler.handleRequest(sql)` which will call the parser (`compileToPinotQuery`) the statement will be parsed again.

Not sure if this is expected. I just noticed it

Reply:
i dont think so but yeah I will take a look to address this as well

Reply:
actually the logic here is a bit more complex than simple reduce the compilation. I will create another PR to follow up.